### ISSUE TITLE
Faulty addition of PauliOp with phase coefficient

### ISSUE THREAD

**Issue by TorbenPeter** [Qiskit/qiskit-terra#7046]
### Information

- **Qiskit Terra version**: 0.18.1
- **Python version**: 3.8.10
- **Operating system**: Ubuntu 20.04.2 LTS

### What is the current behavior?

The addition of two PauliOp produces wrong results when a phase coefficient is used.

### Steps to reproduce the problem

```
import qiskit as qk
from qiskit.opflow.primitive_ops import PauliOp

(PauliOp(qk.quantum_info.Pauli("X")) + PauliOp(qk.quantum_info.Pauli("iY"))).to_matrix()

>>> array([[0.+0.j, 1.-1.j],
       [1.+1.j, 0.+0.j]])
```

### What is the expected behavior?

The expected behaviour can be achieved by applying the phase multiplying with 1j after the fact:

```
(PauliOp(qk.quantum_info.Pauli("X")) + 1j*PauliOp(qk.quantum_info.Pauli("Y"))).to_matrix()

>>>array([[0.+0.j, 2.+0.j],
       [0.+0.j, 0.+0.j]])
```

The deprecated version from aqua also operates correctly:

```
import qiskit as qk
from qiskit.aqua.operators.primitive_ops import PauliOp

(PauliOp(qk.quantum_info.Pauli("X")) + PauliOp(qk.quantum_info.Pauli("iY"))).to_matrix()

>>>array([[0.+0.j, 2.+0.j],
       [0.+0.j, 0.+0.j]])
```

**Comment by 1ucian0** [comment]
This seems solved in main:
```
import qiskit as qk
from qiskit.opflow.primitive_ops import PauliOp

(PauliOp(qk.quantum_info.Pauli("X")) + PauliOp(qk.quantum_info.Pauli("iY"))).to_matrix()
```
```
array([[0.+0.j, 2.+0.j],
       [0.+0.j, 0.+0.j]])

**Comment by 1ucian0** [comment]
Yeap. It seems like this is a https://github.com/Qiskit/qiskit-terra/issues/6815 duplicated and it was solved by #6826. Please, reopen if I'm wrong.
